### ISSUE TITLE
Keyword search ingest module halts with certain images when OCR is enabled

### ISSUE THREAD

**Issue by pyllyukko** [sleuthkit/autopsy#6432]
The keyword search ingest module gets stuck and doesn't complete with certain images and OCR. When the thread/task doesn't complete, the rest of the tasks in the queue will not run so this is effectively a DoS scenario.

Reproduction
------------

Tested with Autopsy 4.16.0.

* Create a new case
* Add [this](https://www.virustotal.com/gui/file/dad2f59d29471ac68bda96e2afaec5809e88b6c9d79e804322a268b579540d7c/detection) single file as data source
* Enable OCR from Tools -> Options -> Keyword Search -> String Extraction -> Enable Optical Character Recognition (OCR)
* Run Keyword Search ingest module
* Observe the keyword search ingest thread/task getting stuck from the Status Area and from Help -> Get Ingest Progress Snapshot

**Comment by rcordovano** [comment]
@pyllyukko, can you provide us with the a copy of the file that is causing the hang? We cannot download it from VirusTotal.

**Comment by pyllyukko** [comment]
> @pyllyukko, can you provide us with the a copy of the file that is causing the hang? We cannot download it from VirusTotal.

Here you go:

![dad2f59d29471ac68bda96e2afaec5809e88b6c9d79e804322a268b579540d7c](https://user-images.githubusercontent.com/1058704/98116688-8a5ec780-1ea0-11eb-90a5-d409eec328b9.png)

**Comment by rcordovano** [comment]
Thanks, @pyllyukko. We'll try to reproduce and investigate the hang here at Basis Technology. We'll keep you posted.

**Comment by rcordovano** [comment]
@pyllyukko, the problem file was being analyzed by tesseract OCR (https://github.com/tesseract-ocr/tesseract) in a process spawned by Autopsy. We had omitted to run the process with one of our "process terminators" that kills a spawned process if the ingest job is cancelled or if a configurable time out occurs. We have added a process terminator and chose a time out of thirty minutes as a compromise between allowing a potentially lengthy OCR process to complete and hanging up an ingest job. 

This mitigation will appear in the Autopsy 4.18.0 release, currently slated for January, 2021.

I have also reported the issue to Tesseract: https://github.com/tesseract-ocr/tesseract/issues/3154.
